### ISSUE TITLE
If creating pre-release with `upload_to_release = True` and using GitHub create pre-release on GitHub

### ISSUE THREAD

**Issue by wintero92** [python-semantic-release/python-semantic-release#544]
## Description

`semantic-release publish --prerelease` should create prerelease on GitHub, i.e., set this [hvcs.py#L201](https://github.com/python-semantic-release/python-semantic-release/blob/2aab9bd4fba532dfa632d2008bd10b2fe1e3eb05/semantic_release/hvcs.py#L201) to `True`

## Use cases

I do not want to tag pre-release as latest or as release at all. It should rather be pre-release.

## Possible implementation

Assuming there is always `-` in pre-release tag this should be enough:
```
    @classmethod
    @LoggedFunction(logger)
    def create_release(cls, owner: str, repo: str, tag: str, changelog: str) -> bool:
        """Create a new release

        https://docs.github.com/rest/reference/repos#create-a-release

        :param owner: The owner namespace of the repository
        :param repo: The repository name
        :param tag: Tag to create release for
        :param changelog: The release notes for this version

        :return: Whether the request succeeded
        """
        try:
            prerelease = False
            if "-" in tag:
                prerelease = True
            Github.session().post(
                f"{Github.api_url()}/repos/{owner}/{repo}/releases",
                json={
                    "tag_name": tag,
                    "name": tag,
                    "body": changelog,
                    "draft": False,
                    "prerelease": prerelease,
                },
            )
            return True
        except HTTPError as e:
            logger.warning(f"Release creation on Github has failed: {e}")
            return False
```

**Comment by bernardcooke53** [comment]
Hey @wintero92 - I actually think this should be part of the prerelease functionality but isn't currently there.

[this call](https://github.com/python-semantic-release/python-semantic-release/blob/a4b625f98da8b4d9364f6dd37260d92a24238cf6/semantic_release/cli.py#L407) should also pass `prerelease` down from the command-line parameters to [`create_release`](https://github.com/python-semantic-release/python-semantic-release/blob/a4b625f98da8b4d9364f6dd37260d92a24238cf6/semantic_release/hvcs.py#L181) - that's a more robust way than checking `if "-" in tag` too. The way this is currently implemented, `prerelease` would get passed to `create_release` regardless of which VCS provider is in use, so that method would need to be changed on all of those classes, not just `GitHub`.

Thanks for pointing this out - I'd be happy to look at a PR if you have time to put one together

**Comment by bernardcooke53** [comment]
Fixed by #587
